reformat, +pytest, +solution (#9)
Ada-Developers-Academy/core-pses: 07/reshape-a-matrix.checkpoint.md

##### !question

Implement `reshape_matrix`.

Use these tests to guide your solution. These tests may make different assumptions about the problem than you did! Do not alter your assumptions above, and take these assumptions for this challenge.

```py
def test_convert_2_2_to_1_4()
    matrix = [[1,2],[3,4]]
    r = 1
    c = 4

    reshaped_matrix = reshape_matrix(matrix, r, c)

    assert reshaped_matrix == [[1,2,3,4]]

> def test_cannot_to_reshape()

Reviewer: the name seems just a little weird to me
```suggestion
def test_cannot_reshape()
```

> Write the code to solve the problem here.  You can use the tests to guide your solution.

Reviewer 2: I'm suggesting this because:

1. "Write the code to solve the problem" felt a lil awkward to me
2. We discussed how students will need to change their assumptions. It'll be good to acknowledge to them that they should do that, and it doesn't invalidate their previous thoughts.

```suggestion
Implement `reshape_matrix`.

Use these tests to guide your solution. These tests may make different assumptions about the problem than you did! Do not alter your assumptions above, and take these assumptions for this challenge.
```

```py
import unittest

> from main import *

Reviewer: is all we need reshape_matrix? * is ok, but generally recommended to be avoided. not a big deal since there probably won't be any other stray symbols to import, but something to consider.
```suggestion
from main import reshape_matrix
```

> import numpy as np

Reviewer: saw this mentioned in another review, that if we aren't using numpy, we should be able to remove this
```suggestion
```
